### ISSUE TITLE
Beta tags and commit hashes don't work

### ISSUE THREAD

**Issue by fregante** [biomejs/pre-commit#29]
The readme states:

https://github.com/biomejs/pre-commit/blob/3957a819ba892359fd4c46a6ad636b8f6e0fef75/README.md?plain=1#L23-L24

But I'm unable to use it

### Tag

https://github.com/biomejs/biome/releases/tag/cli%2Fv1.9.5-nightly.81fdedb

```yml

  - repo: https://github.com/biomejs/pre-commit
    rev: cli/v1.9.5-nightly.81fdedb
```
```
[INFO] Initializing environment for https://github.com/biomejs/pre-commit.
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'checkout', 'cli/v1.9.5-nightly.81fdedb')
return code: 1
stdout: (none)
stderr:
    error: pathspec 'cli/v1.9.5-nightly.81fdedb' did not match any file(s) known to git
```

### Hash

https://github.com/biomejs/biome/tree/81fdedb

```yml
  - repo: https://github.com/biomejs/pre-commit
    rev: 81fdedb
```

```
[INFO] Initializing environment for https://github.com/biomejs/pre-commit.
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'checkout', '81fdedb')
return code: 1
stdout: (none)
stderr:
    error: pathspec '81fdedb' did not match any file(s) known to git
```

**Comment by fregante** [comment]
`v2.0.0-beta.1` works now that it exists, but I think that either the docs need to be updated or the handling of `rev` needs to be fixed to support hashes, etc

**Comment by grst** [comment]
I would rather not have beta releases of `biomejs/pre-commit`. This is because the pre-commit autoupdate service always updates to the latest tag without making any effort to parse the version. This means every repo relying on that service will get pull requests updating to beta tags instead of the latest stable version, e.g. https://github.com/scverse/anndata-plot/pull/10#event-17301946078

CC @flying-sheep

**Comment by flying-sheep** [comment]
An alternative would be https://github.com/biomejs/biome-vscode/issues/593

Then we could update our pre-commit config to the beta release while everything keeps working. But one of the two is really necessary (both would be ideal)
